client/src/components/Dashboard.js

```diff
@@ -1,70 +1,67 @@
 
-function Dashboard() {
-    const navigate = useNavigate();
-    const [bills, setBills] = useState();
+import { connect } from 'react-redux';
+import { loadUser } from '../actions/auth';
 
-    useEffect(() => {
-        const token = localStorage.getItem('token');
-        
-        // Check to see if there's a token.
-        // Then get all the bills for that user.
-        if(token) {
-            const {user} = jwt_decode(token);
-            if(!user) {
-                localStorage.removeItem('token');
-                navigate('/login');
-            }
-            
-            getBills(token, user.id);
-        } else  {
-            navigate('/login');
-        }
+function Dashboard({ loadUser }) {
+	const navigate = useNavigate();
+	const [bills, setBills] = useState();
 
-        // Query the backend and database to get all the bills.
-        async function getBills(token, userid) {
-            const url = `http://localhost:8000/app/bill/?userid=${userid}`;
-            const headers = { 'x-auth-token': token};
-            const res = await axios.get(url, { headers });
+	useEffect(() => {
+		const token = localStorage.getItem('token');
 
-            if(res.error) {
-                console.log(res.error);
-            } else {
-                console.log(res.data.bills);
-                setBills(res.data.bills);
-            }
+		// Check to see if there's a token.
+		// Then get all the bills for that user.
+		if (token) {
+			loadUser(token);
+			const { user } = jwt_decode(token);
+			if (!user) {
+				localStorage.removeItem('token');
+				navigate('/login');
+			}
 
-            // if(res.bills?.length === 0) {
-            //     console.log('You have no bills at the moment');
-            // } else {
-            //     setBills(...res.bills);
-            // }
-            
-        };
+			getBills(token, user.id);
+		} else {
+			navigate('/login');
+		}
 
-    }, []);
+		// Query the backend and database to get all the bills.
+		async function getBills(token, userid) {
+			const url = `http://localhost:8000/app/bill/?userid=${userid}`;
+			const headers = { 'x-auth-token': token };
+			const res = await axios.get(url, { headers });
 
-    return (
-        <div>
-        
-            {/* navigate is a function that can be used to redirect user. */}
-            {/* <button onClick={() => navigate('/login')}>
-                Click to go back to the log in page.
-            </button> */}
-
-            {/* Navigate can be used to check if the user is authed. If not, use it to redirect user back. */}
-            {/* <Navigate to='/login' /> */}
+			if (res.error) {
+				console.log(res.error);
+			} else {
+				console.log(res.data.bills);
+				setBills(res.data.bills);
+			}
 
-            Welcome back!
+			// if(res.bills?.length === 0) {
+			//     console.log('You have no bills at the moment');
+			// } else {
+			//     setBills(...res.bills);
+			// }
+		}
+	}, [loadUser, navigate]);
```

Comment:
added this in the useEffect, to fix the errors

`WARNING in src\components\Dashboard.js
[1]   Line 51:5:  React Hook useEffect has missing dependencies: 'loadUser' and 'navigate'. Either include them or remove the dependency array. If 'loadUser' changes too often, find 
the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps`